src/Sylius/Bundle/ApiBundle/StateProcessor/CountryProcessor.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProcessor;
+
+use ApiPlatform\Metadata\DeleteOperationInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use Sylius\Bundle\ApiBundle\Exception\ProvinceCannotBeRemoved;
+use Sylius\Component\Addressing\Checker\CountryProvincesDeletionCheckerInterface;
+
+final class CountryProcessor implements ProcessorInterface
+{
+    public function __construct(
+        private ProcessorInterface $persistProcessor,
+        private ProcessorInterface $removeProcessor,
+        private CountryProvincesDeletionCheckerInterface $countryProvincesDeletionChecker,
+    ) {
+    }
+
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = [])
+    {
+        if ($operation instanceof DeleteOperationInterface) {
+            return $this->removeProcessor->process($data, $operation, $uriVariables, $context);
+        }
+
+        if (!$this->countryProvincesDeletionChecker->isDeletable($data)) {
```

Comment:
@TheMilek I'm not sure I'm reading right, but you're allowing to delete country before checking if country is deletable.

Reply:
@diimpp, In fact, country deletion is dead code, because if I see correctly, we don't support it either in the API or in the admin panel. 
The `CountryProvincesDeletionChecker` service is triggered during a PUT operation